doc/manual/source/glossary.md

```diff
@@ -31,9 +31,24 @@
 
   The industry term for storage and retrieval systems using [content addressing](#gloss-content-address). A Nix store also has [input addressing](#gloss-input-addressed-store-object), and metadata.
 
+- [derivation]{#gloss-derivation}
+
+  A derivation can be thought of as a [pure function](https://en.wikipedia.org/wiki/Pure_function) that produces new files from existing files, taking into account how these files refer to each other.
+
+  In Nix, linked files are represented as [store objects][store object].
```

Comment:
Not clear what is meant by "linked files".

Reply:
That's a short way of saying "files that refer to each other". It is deliberately left a bit vague to avoid using "reference" and skipping ahead of "store object".

But maybe I'm being overly cautious at an undue expense of clarity, especially given the jungle of links to other terms, so it may as well say "files with references".

Reply:
How about this?

```suggestion
  A derivation can be thought of as a [pure function](https://en.wikipedia.org/wiki/Pure_function) that produces new [store objects][store object] from existing store objects.
```